nntrainer/tensor/tensor.cpp

```diff
@@ -718,10 +718,14 @@ Tensor &Tensor::dot(Tensor const &m, Tensor &result, bool trans, bool trans_m,
     throw exception::not_supported("Error: support only for rank of dot "
                                    "matrix <= 2");
   }
-  if (trans && dim.rank() > 2) {
-    throw exception::not_supported("Error: support only for rank of dot "
-                                   "matrix <= 2 with trans");
-  }
+
+  // Comment out with intension to support the calculation wrt. batch and height
+  // direction of this tensor. It is OK as long as m is 2D
```

Comment:
in this case, `m` is treated as a 2D matrix. Can we keep the check and let the caller reshape before the call?

Reply:
in a some sense, the reshape is kind of redundant. do we really need to? A x B = C  A could be batch, channel, height, width.. it will store.. like batch, channel, width*height. and B is 2D ( height_A , width ) .. I guess it Is good to go
as long as the width is matched.

Reply:
then how about just give a warning. This is just to ensure that the caller does not make a mistake.

Reply:
done.